src/overlays/actors/ovl_En_Trt2/z_en_trt2.c

```diff
@@ -76,111 +103,832 @@ static DamageTable D_80AD58D8 = {
+
+    sp40 = player->actor.world.pos;
+
+    if (this->unk_3B2 == 8) {
+        sp40.y = player->bodyPartsPos[7].y + 3.0f;
+    } else {
+        sp40.y = player->bodyPartsPos[7].y + 45.0f;
+    }
+
+    sp34 = this->actor.world.pos;
+
+    Math_SmoothStepToS(&this->unk_3D4, Math_Vec3f_Pitch(&sp34, &sp40), 4, 0x1C70, 1);
+    this->unk_3D4 = CLAMP(this->unk_3D4, -0x1C70, 0x1C70);
+}
+
+void func_80AD5394(s16 arg0, s16 arg1, Vec3f* arg2, Vec3s* arg3, s32 arg4) {
+    Vec3f sp7C;
+    Vec3f sp70 = gZeroVec3f;
+    Vec3s sp68;
+    MtxF sp28;
+
+    Matrix_MultiplyVector3fByState(&sp70, &sp7C);
+    Matrix_CopyCurrentState(&sp28);
+    func_8018219C(&sp28, &sp68, 0);
+
+    *arg2 = sp7C;
+
+    if (arg4) {
+        sp68.x += arg0;
+        sp68.y += arg1;
+        Math_SmoothStepToS(&arg3->x, sp68.x, 4, 0x1FFE, 1);
+        Math_SmoothStepToS(&arg3->y, sp68.y, 4, 0x1FFE, 1);
+        Math_SmoothStepToS(&arg3->z, sp68.z, 4, 0x1FFE, 1);
+    } else {
+        arg3->x = sp68.x;
+        arg3->y = sp68.y;
+        arg3->z = sp68.z;
+    }
+}
+
+s32 EnTrt2_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                            Actor* thisx) {
+    EnTrt2* this = THIS;
+
+    if ((limbIndex == 8) || (limbIndex == 13) || (limbIndex == 19)) {
+        rot->y += (s16)Math_SinS(this->unk_33C[limbIndex]) * 200;
+        rot->z += (s16)Math_CosS(this->unk_372[limbIndex]) * 200;
+    }
+    return false;
+}
+
+void EnTrt2_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnTrt2* this = THIS;
+    Vec3f sp30 = { 0.0f, -30.0f, 0.0f };
+    s32 phi_v0 = false;
+
+    if (this->unk_3B2 >= 8) {
+        phi_v0 = true;
+    }
+
+    if (limbIndex == 21) {
+        func_80AD5394(this->unk_3D4, this->unk_3D6, &this->unk_3C8, &this->unk_3C2, phi_v0);
+        Matrix_InsertTranslation(this->unk_3C8.x, this->unk_3C8.y, this->unk_3C8.z, MTXMODE_NEW);
+        Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+        Matrix_RotateY(this->unk_3C2.y, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(this->unk_3C2.x, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(this->unk_3C2.z, MTXMODE_APPLY);
+        Matrix_MultiplyVector3fByState(&sp30, &this->actor.focus.pos);
+    }
+}
+
+void EnTrt2_UnkDraw(GlobalContext* globalCtx, s32 limbIndex, Actor* thisx) {
+    EnTrt2* this = THIS;
+
+    if (limbIndex == 21) {
+        Matrix_InsertTranslation(this->unk_3C8.x, this->unk_3C8.y, this->unk_3C8.z, MTXMODE_NEW);
+        Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+        Matrix_RotateY(this->unk_3C2.y, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(this->unk_3C2.x, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(this->unk_3C2.z, MTXMODE_APPLY);
+    }
+}
+
+void func_80AD56E8(Actor* thisx, GlobalContext* globalCtx) {
+    static UNK_PTR D_80AD5978[] = {
```

Comment:
```suggestion
    static TEXTURE_PTR D_80AD5978[] = {
```

Reply:
It's `TexturePtr`, but yeah, please change it

Reply:
Oops, my bad. Fixed in the suggestion.